File: JULY-2025/07-07-2025/p1353.swift
class Solution {
    func maxEvents(_ events: [[Int]]) -> Int {
        let sorted = events.sorted { $0[1] < $1[1] }
        let maxDay = sorted.map { $0[1] }.max() ?? 0
        var attended = [Bool](repeating: false, count: maxDay + 1)
    
        var count = 0
        for event in sorted {
            for day in event[0]...event[1] {
                if !attended[day] {
                    attended[day] = true
                    count += 1
                    break
                }
            }
        }
        return count
    }
}

File: JULY-2025/07-07-2025/p1353.dart
import 'package:collection/priority_queue.dart';
import 'dart:math'; 

class Solution {
  int maxEvents(List<List<int>> events) {

    events.sort((a, b) => a[0].compareTo(b[0]));

    final PriorityQueue<int> minHeap = PriorityQueue<int>((a, b) => a.compareTo(b));

    int attendedEvents = 0;
    int eventIndex = 0; // Pointer to the next event to consider from the sorted list
    int n = events.length;
    int maxEndDay = 0;

    if (n > 0) {
      for (var event in events) {
        maxEndDay = max(maxEndDay, event[1]);
      }
    } 
    else {
      return 0; // No events, so 0 attended.
    }

    for (int day = 1; day <= maxEndDay; day++) {
   
      while (eventIndex < n && events[eventIndex][0] == day) {
        minHeap.add(events[eventIndex][1]); // Add the event's end day
        eventIndex++;
      }

      while (minHeap.isNotEmpty && minHeap.first < day) {
        minHeap.removeFirst(); // Use removeFirst() for PriorityQueue from 'collection'
      }

      if (minHeap.isNotEmpty) {
        minHeap.removeFirst(); // Attend this event (remove it from consideration)
        attendedEvents++;
      }

      // Optimization
      if (eventIndex == n && minHeap.isEmpty) {
        break;
      }
    }

    return attendedEvents;
  }
}